proposals/extensions.md

```diff
@@ -399,54 +399,237 @@ These requirements need more refinement as implementation progresses, and may ne
+it doesn't matter what is the name of the extension grouping type, what matters is the signature of the extension block it belongs to. The public API signature should not be seen as
+`Extension.<>E__ExtensionGroupingTypeNameForObject.M(System.Int32)` but rather as `Extension.extension(object).M(int)`. In other words, extension members should be seen as having two sets of
+type parameters and two sets of parameters.
+
+#### Overview
+
+Extension blocks are grouped by their CLR-level signature. Each CLR equivalency group is emitted as an **extension grouping type** with a content-based name.
+Extension blocks within a CLR equivalency group are then sub-grouped by C# equivalency. Each C# equivalency group is emitted as an **extension marker type** with a content-based name, nested in its corresponding extension grouping type.
+An extension marker type contains a single **extension marker method** which encodes an extension parameter.
+Each extension member is emitted with a skeleton body in the right extension grouping type, refers back to an extension marker type by its name via an attribute, and is accompanied by a top-level static **implementation method** with a modified signature.    
+
+Here's a schematized overview of metadata encoding:
+```
+[Extension]
+static class EnclosingStaticClass
+{
+    [Extension]
+    public sealed class ExtensionGroupingType1 // has type parameters with minimal constraints sufficient to keep extension member declarations below valid
+    {
+        private static class ExtensionMarkerType1 // has re-declared type parameters with full fidelity of C# constraints
+        {
+            private static void <Extension>$(... extension parameter ...) // extension marker method
+        }
+        ... ExtensionMarkerType2, etc ...
+
+        ... extension members for ExtensionGroupingType1, each points to its corresponding extension marker type ...
+    }
+
+    ... ExtensionGroupingType2, etc ...
+
+    ... implementation methods ...
+}
+```
+
+The enclosing static class is emitted with an `[Extension]` attribute.  
+
+
+### CLR-level signature vs. C#-level signature
+
+The CLR-level signature of an extension block results from:
+- normalizing type parameter names to `T0`, `T1`, etc ...
+- removing attributes
+- erasing the parameter name
+- erasing parameter modifiers (like `ref`, `in`, `scoped`, ...)
+- erasing tuple names
+- erasing nullability annotations
+- erasing `notnull` constraints
+
+Note: other constraints are preserved, such as `new()`, `struct`, `class`, `allows ref struct`, `unmanaged`, and type constraints.
+
+#### Extension grouping types
+
+An extension grouping type is emitted to metadata for each set of extension blocks in source with a the same CLR-level signature (extension grouping types are an evolution of the previous "skeleton type" design).  
```

Comment:
>(extension grouping types are an evolution of the previous "skeleton type" design) [](http://example.com/codeflow?start=129&length=82)

Consider dropping this note. I do not find it particularly helpful to a reader of the spec.